File: src/org/mapsforge/map/layer/renderer/Shape.java
package org.mapsforge.map.layer.renderer;

import org.jetbrains.annotations.NotNull;
import org.mapsforge.core.graphics.Paint;
import org.mapsforge.core.graphics.TileBitmap;
import org.mapsforge.core.model.Point;

public interface Shape extends TileBitmap {
  void drawPolyLine(@NotNull Point[][] coordinates, @NotNull Point origin, float dy);

  boolean beginFillOrSetLineStyle(@NotNull Paint paint);

  void endFill();

  void drawCircle(double x, double y, float radius);

  void drawTextRotated(@NotNull String text, @NotNull Point start, @NotNull Point end, @NotNull Point origin, Paint paintBack);
}

File: mapsforge-ex/src/Shape.kt
package org.mapsforge.map.layer.renderer

import org.mapsforge.core.graphics.Bitmap
import org.mapsforge.core.graphics.Paint
import org.mapsforge.core.graphics.TileBitmap
import org.mapsforge.core.model.Point

public trait Shape : TileBitmap {
  public fun drawPolyLine(coordinates: Array<Array<Point>>, origin: Point, dy: Float): Unit

  public fun beginFillOrSetLineStyle(paint: Paint): Boolean

  public fun endFill(): Unit

  public fun drawCircle(x: Double, y: Double, radius: Float): Unit

  public fun drawTextRotated(text: String, start: Point, end: Point, origin: Point, paintFront: Paint): Unit

  public fun drawText(text: String, x: Double, y: Double, paintFront: Paint): Unit

  public fun drawSymbol(symbol: Bitmap, x: Double, y: Double, rotation: Float): Unit
}
